### ISSUE TITLE
[Question] CrawlerTrait::call() works only internally

### ISSUE THREAD

**Issue by shehi** [laravel/framework#9698]
I have been trying to write a unit-test for my OAuth implementation these past 2 days and regardless what I tried, all external ```call()```s ended up rotating inside my app, so I started to dig up. Finally my tracing led me to where ```CrawlerTrait::call()``` is called, which in turn calls ```Kernel::handle()``` method, where it becomes interesting: the latter, utilizes ```Kernel::sendRequestThroughRouter()``` method which passes every request through our own app's routing rules - even calls to "http://google.com" becomes a ```RedirectResponse``` object with funny redirection to "http://google.com/en/home" (```en/home``` is an internal localized redirection in my app), never reaching Google's website.

So question is: Is ```CrawlerTrait``` intended only for internal crawling, never reaching outside world? If so, why wasn't this small but important fact not mentioned in docs, which practically wastes hours if not days of precious time? Or have I missed that piece when I read docs? And finally: if the answer to my first question is Yes, then I guess I will have to use something else, like cURL or GuzzleHttp library etc, am I correct?

Thanks.

**Comment by shehi** [comment]
Also, to be honest, I first encountered this problem when I created my custom [```LocalizedRouter```](https://github.com/AudithSoftworks/Basis/blob/translation/app/Services/LocalizedRouter.php) for my open-source app, which extends Laravel's ```Router``` class. Also added special ```missingMethod()``` handling to my parent ```Controller``` class to handle those localized routes (as in, a route such as ```/home``` gets localized/aliased as ```/ev``` - in Azerbaijani - and that also gets resolved by controller so that ```HomeController@getHome()``` is being called. Thing is, my routing and controllers and everything work very well, but when it comes to testing them with built-in ```CrawlerTrait``` the situation becomes so messed up, I can't make heads from tails - I can't even reach ```/home``` route in tests, always get redirected to ```/login``` route instead. Still trying to figure out the why of it.

So overall, we have two problems of the same source: ```CrawlerTrait``` doesn't work as it should and tries to do some extra stuff behind the curtain to accommodate some strange purpose.
